release/internal/pinnedversion/pinnedversion.go

```diff
@@ -47,167 +48,222 @@ var noImageComponents = []string{
+	Components     map[string]registry.Component `yaml:"components"`
 }
 
-// PinnedVersionData represents the data needed to generate the pinned version file from the template.
-type PinnedVersionData struct {
-	// ReleaseName is the name of the release.
-	ReleaseName string
+func NewCalicoVersionData(productVersion, operatorVersion string) *CalicoPinnedVersionData {
+	return &CalicoPinnedVersionData{
+		calico:   version.New(productVersion),
+		operator: version.New(operatorVersion),
+	}
+}
 
-	// BaseDomain is the base domain for the docs site.
-	BaseDomain string
+type CalicoPinnedVersionData struct {
+	calico   version.Version
+	operator version.Version
+}
 
-	// ProductVersion is the version of the product.
-	ProductVersion string
+func (v *CalicoPinnedVersionData) ProductVersion() string {
+	return v.calico.FormattedString()
+}
 
-	// Operator is the operator component.
-	Operator registry.Component
+func (v *CalicoPinnedVersionData) OperatorVersion() string {
+	return v.operator.FormattedString()
+}
 
-	// Note is the note for the release.
-	Note string
+func (v *CalicoPinnedVersionData) HelmChartVersion() string {
+	return v.calico.FormattedString()
+}
 
-	// Hash is the hash of the release.
-	Hash string
+func (v *CalicoPinnedVersionData) Hash() string {
+	return fmt.Sprintf("%s-%s", v.calico.FormattedString(), v.operator.FormattedString())
+}
 
-	// ReleaseBranch is the release branch of the release.
-	ReleaseBranch string
+func (v *CalicoPinnedVersionData) ReleaseBranch(releaseBranchPrefix string) string {
+	return fmt.Sprintf("%s-%s", releaseBranchPrefix, v.calico.Stream())
 }
 
-// PinnedVersion represents an entry in pinned version file.
-type PinnedVersion struct {
-	Title          string                        `yaml:"title"`
-	ManifestURL    string                        `yaml:"manifest_url"`
-	ReleaseName    string                        `yaml:"release_name"`
-	Note           string                        `yaml:"note"`
-	Hash           string                        `yaml:"full_hash"`
-	TigeraOperator registry.Component            `yaml:"tigera-operator"`
-	Components     map[string]registry.Component `yaml:"components"`
+// calicoTemplateData is used to generate the pinned version file from the template.
+type calicoTemplateData struct {
+	ReleaseName    string
+	BaseDomain     string
+	ProductVersion string
+	Operator       registry.Component
+	Note           string
+	Hash           string
+	ReleaseBranch  string
 }
 
-// PinnedVersionFile represents the pinned version file.
-type PinnedVersionFile []PinnedVersion
+func (d *calicoTemplateData) ReleaseURL() string {
+	return fmt.Sprintf("https://%s.%s", d.ReleaseName, d.BaseDomain)
+}
 
 func pinnedVersionFilePath(outputDir string) string {
 	return filepath.Join(outputDir, pinnedVersionFileName)
 }
 
-func operatorComponentsFilePath(outputDir string) string {
-	return filepath.Join(outputDir, operatorComponentsFileName)
+type CalicoPinnedVersions struct {
```

Comment:
Confusing to have structs named `CalicoPinnedVersions` and `CalicoPinnedVersionData` in the same file without any comments explaining the separation - what is the intended scope of these two types?